src/vcpkg/base/messages.cpp

```diff
@@ -243,29 +242,49 @@ namespace vcpkg::msg
         }
     }
 
-    static std::string locale_file_name(StringView language)
-    {
-        std::string filename = "messages.";
-        filename.append(language.begin(), language.end()).append(".json");
-        return filename;
-    }
-
-    void threadunsafe_initialize_context(const Filesystem& fs, StringView language, const Path& locale_base)
+    ExpectedS<Json::Object> get_message_map_from_lcid(int LCID)
     {
         threadunsafe_initialize_context();
+        auto embedded_filesystem = cmrc::cmakerc::get_filesystem();
 
-        auto path_to_locale = locale_base / locale_file_name(language);
+        const std::string locale_path = get_locale_path(LCID);
 
-        auto message_map = Json::parse_file(VCPKG_LINE_INFO, fs, path_to_locale);
-        if (!message_map.first.is_object())
+        auto file = embedded_filesystem.open(locale_path);
+
+        auto maybe_map = Json::parse_object(StringView{file.begin(), file.end()}, locale_path);
+
+        return maybe_map;
+    }
+
+    std::string get_locale_path(int LCID) { return fmt::format("locales/messages.{}.json", get_language_tag(LCID)); }
+
+    StringLiteral get_language_tag(int LCID)
+    {
+        std::array<std::pair<int, StringLiteral>, 14> languages = {std::pair<int, StringLiteral>(1029, "cs"),
```

Comment:
Please add a link to the documentation where you got this LCID codes.
The only thing I could find was this: https://learn.microsoft.com/en-us/visualstudio/ide/reference/lcid-devenv-exe?view=vs-2022

Reply:
I found this link. Do you think this is better? @vicroms 
https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2015/ide/reference/lcid-devenv-exe?view=vs-2015&redirectedfrom=MSDN